openmdao/docs/theory_manual/setup_stack.rst

```diff
@@ -0,0 +1,120 @@
+.. _theory_setup_stack:
+
+*********************************************************************
+The System Setup Stack: Understanding When to Use setup and configure
+*********************************************************************
+
+This document explains what happens during the OpenMDAO setup process, and how some of the model
+API methods interact during that process.
+
+The purpose of the setup process is to prepare the data structures that OpenMDAO need to efficiently
+run your model or driver. In particular, this includes setting up the vectors used for passing data
+to inputs, converging solvers, and calculating derivatives. It also includes setting up the MPI
+communicators.
+
+Setup also performs some level of model checking, mainly for critical errors. More extensive model
+checking can be done by setting "check" to True when calling `setup`, or by using the :ref:`openmdao command
+line check<om-command>`. It is recommend that you do this before your first full run of the model.
+
+The OpenMDAO `Group` API includes three methods that are invoked during the setup process: `setup`, `configure`, and
+`initialize`. Most of the time, setup is all you need to build a group. The specific use case for
+configure is shown below in the next section. The configure method is only used for declaring options for your
```

Comment:
sounds like this is referring to `initialize` not `configure`... from `System`:
        self._declare_options()
        self.initialize()
        self.options.update(kwargs)